rotel_python_processor_sdk/rotel_sdk/README.md

```diff
@@ -125,32 +215,371 @@ def process_logs(resource_logs: ResourceLogs):
+    ]
+    data_point.attributes = attributes
+
+    # Set the scale - using scale=0 means base=2, giving us good resolution
+    # For latency measurements in milliseconds
+    data_point.scale = 0  # base = 2^(2^0) = 2
+
+    # Total count of all measurements
+    data_point.count = 10000  # 10k requests in this time window
+
+    # Sum of all latency values in milliseconds
+    # Most requests 5-20ms, some 100-500ms, few >1000ms
+    # Rough calculation: 9500*12ms + 450*200ms + 50*1500ms = 264,000ms
+    data_point.sum = 264000.0  # Total latency in milliseconds
+
+    # Zero count - requests that took essentially 0ms (very rare for HTTP)
+    data_point.zero_count = 5
+
+    # Create positive buckets for the main distribution
+    positive_buckets = ExponentialHistogramBuckets()
+
+    # With scale=0 (base=2), bucket boundaries are:
+    # Bucket index i covers range (2^i, 2^(i+1)]
+    # For milliseconds:
+    # index 2: (4ms, 8ms]     - some fast requests
+    # index 3: (8ms, 16ms]    - most normal requests  
+    # index 4: (16ms, 32ms]   - slower normal requests
+    # index 5: (32ms, 64ms]   - getting slower
+    # index 6: (64ms, 128ms]  - slow requests
+    # index 7: (128ms, 256ms] - very slow requests
+    # index 8: (256ms, 512ms] - problematic requests
+    # index 9: (512ms, 1024ms] - very problematic
+    # index 10: (1024ms, 2048ms] - extremely slow
+
+    positive_buckets.offset = 2  # Start from bucket index 2 (4-8ms range)
+
+    # Bucket counts representing our latency distribution
+    positive_buckets.bucket_counts = [
+        850,  # index 2: (4ms, 8ms] - 850 requests
+        4200,  # index 3: (8ms, 16ms] - 4200 requests (peak)
+        3800,  # index 4: (16ms, 32ms] - 3800 requests 
+        600,  # index 5: (32ms, 64ms] - 600 requests
+        200,  # index 6: (64ms, 128ms] - 200 requests
+        150,  # index 7: (128ms, 256ms] - 150 requests
+        100,  # index 8: (256ms, 512ms] - 100 requests
+        60,  # index 9: (512ms, 1024ms] - 60 requests
+        30,  # index 10: (1024ms, 2048ms] - 30 requests (>1 second!)
+        10,  # index 11: (2048ms, 4096ms] - 10 very slow requests
+    ]
+
+    data_point.positive = positive_buckets
+
+    # No negative buckets for latency (can't have negative time)
+    data_point.negative = None
+
+    # Add some exemplars to show specific slow requests
+    exemplars = []
+
+    # Exemplar 1: A normal request
+    exemplar1 = Exemplar()
+    exemplar1.time_unix_nano = current_time_nano - (30 * 1_000_000_000)  # 30 seconds ago
+    exemplar1.value = ExemplarValue.AsDouble(12.5)  # 12.5ms - typical request
+    exemplar1.filtered_attributes = [
+        KeyValue(key="user.id", value=AnyValue("user123")),
+        KeyValue(key="request.id", value=AnyValue("req-normal-456"))
+    ]
+    # Add trace context for this exemplar
+    exemplar1.trace_id = b'\x01\x02\x03\x04\x05\x06\x07\x08\x09\x0a\x0b\x0c\x0d\x0e\x0f\x10'
+    exemplar1.span_id = b'\x11\x12\x13\x14\x15\x16\x17\x18'
```

Comment:
How would a user typically set these values to actual trace/span ids?

Reply:
Oh yeah, if you haven't checked them out yet it's worth googling. The SDK provides them either by a filter or a reservoir (for tail sampling) iirc. This example is a bit contrived as it's a demonstration on exercising the API. However your question is interesting because theoretically if rotel is deployed side by side with an application you could theoretically use additional libraries to store and fetch some additional data to use as exemplars.